browser/brave_wallet/brave_wallet_provider_impl_unittest.cc

```diff
@@ -195,41 +225,193 @@ TEST_F(BraveWalletProviderImplUnitTest,
+  EXPECT_EQ(callback_called, true);
+}
+
+TEST_F(BraveWalletProviderImplUnitTest, AddUnapproved1559TransactionError) {
+  // This makes sure the state manager gets the chain ID
+  browser_task_environment_.RunUntilIdle();
+
+  // We don't need to check every error type since that is checked by
+  // eth_tx_controller_unittest but make sure an error type is handled
+  // correctly.
+  bool callback_called = false;
+  NavigateAndAddEthereumPermission();
+  provider()->AddUnapproved1559Transaction(
+      mojom::TxData1559::New(
+          // Can't specify both gas price and max fee per gas
+          mojom::TxData::New("0x00", "0x01", "0x00",
+                             "0xbe862ad9abfe6f22bcb087716c7d89a26051f74c",
+                             "0x00", std::vector<uint8_t>()),
+          "0x04", "0x0", "0x0"),
+      from(),
+      base::BindLambdaForTesting([&](bool success, const std::string& id,
+                                     const std::string& error_message) {
+        EXPECT_FALSE(success);
+        EXPECT_TRUE(id.empty());
+        EXPECT_FALSE(error_message.empty());
+        callback_called = true;
+      }));
+  browser_task_environment_.RunUntilIdle();
+  EXPECT_EQ(callback_called, true);
+}
+
 TEST_F(BraveWalletProviderImplUnitTest,
-       OnAddEthereumChainRequestCompletedSuccess) {
-  brave_wallet::EthJsonRpcController controller(shared_url_loader_factory(),
-                                                prefs());
-  BraveWalletProviderImpl provider_impl(
-      controller.MakeRemote(),
-      std::make_unique<brave_wallet::BraveWalletProviderDelegateImpl>(
-          web_contents(), web_contents()->GetMainFrame()),
-      prefs());
-  int callback_is_called = 0;
-  provider_impl.AddEthereumChain(
-      R"({"params": [{
-        "chainId": "0x111",
-        "chainName": "Binance1 Smart Chain",
-        "rpcUrls": ["https://bsc-dataseed.binance.org/"]
-      }]})",
-      base::BindLambdaForTesting(
-          [&callback_is_called](bool success, int error_code,
-                                const std::string& error_message) {
-            base::Value empty;
-            EXPECT_EQ(error_code, 0);
-            ASSERT_TRUE(error_message.empty());
-            callback_is_called++;
-          }));
-  EXPECT_EQ(callback_is_called, 0);
-  provider_impl.OnAddEthereumChainRequestCompleted("0x111", "");
-  EXPECT_EQ(callback_is_called, 1);
-  provider_impl.OnAddEthereumChainRequestCompleted("0x111", "");
-  EXPECT_EQ(callback_is_called, 1);
+       AddUnapproved1559TransactionNoPermission) {
+  // This makes sure the state manager gets the chain ID
+  browser_task_environment_.RunUntilIdle();
+
+  // We don't need to check every error type since that is checked by
+  // eth_tx_controller_unittest but make sure an error type is handled
+  // correctly.
+  bool callback_called = false;
+  provider()->AddUnapproved1559Transaction(
+      mojom::TxData1559::New(
+          // Can't specify both gas price and max fee per gas
```

Comment:
Shouldn't we use a valid txdata here to check no permission case only?

Reply:
Fixed in d8c9c09e62fadd786f09bfdf932a2573f3dd5358